### DISCUSSION TITLE
Async calls

### DISCUSSION THREAD

**Discussion by MalteMagnussen** [netbox-community/pynetbox#629] [Q&A]
Is there a way to do async non-blocking calls to netbox with this lib?

I want to do a bunch of async calls out to netbox, **and** other systems at the same time, instead of sequentially.

**Comment by red-tux** [comment]
I got around this by using thread and process pools.

```
def get_circuits(workers=10,filter={}):
    with Pool(processes=workers) as executor:
    # with ThreadPoolExecutor(max_workers=workers) as executor:
    # Use map to apply the function to each item in the list
        results = list(executor.map(parse_circuit, nb.circuits.circuits.filter(**filter)))
 ```
 
 The code as shown will use Process Pools, but as you can see the method of using Thread Pools is the same.  The `parse_circuit` just takes the next result from the filter, parses it and returns it, which is subsequently added to the results list.

**Reply by MalteMagnussen** [reply]
Thanks for answering.

I ended up writing my own netbox client that is async with proper typing using pydantic.

**Reply by red-tux** [reply]
Is that code available for others?  I see there is an async library for netbox that heavily leverages swagger for the endpoint names.
https://github.com/vapor-ware/aionetbox

**Reply by MalteMagnussen** [reply]
No, sorry. It was written on company time, and is therefore property of my workplace. It also has too much hardcoding for our specific setup to be useful to others.
